src/gpio.rs

```diff
@@ -1394,6 +1552,14 @@ mod chip {
         None, None, None, None, None, None, None, None, None, None,
     ];
 
+    #[allow(clippy::type_complexity)]
+    #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "nightly"))]
+    pub(crate) static mut PIN_NOTIFIERS: [Option<Box<Notification>>; 40] = [
```

Comment:
No need for `Box`-ing here. I think all other places in the `gpio` module where `Box` is used were protected with the `alloc` feature, but given that we do not need to `Box` here (unlike in the other case from where you copied, where we DO need to do it because we are keeping around `FnMut` instances, and each of these has a distinct type) we can happily get by without it here.